docs/operational.md

```diff
@@ -0,0 +1,66 @@
+# Operational docs
+## tips and tricks for running this thing IRL
+
+
+### Things to keep in mind
+
+Huey pipelines are linked-list tasks. This means that when Huey pulls a task out of the queue,
+and that task is part of a pipeline, the whole pipeline is popped from the queue. Because of this,
+if the worker consuming a task terminates without pushing the pipeline back to the queue, Huey loses
+track of the pipeline. This lead to the creation of the `scan_for_stalled_pipelines` job, which runs
+on a cron schedule. It looks for pipelines that have not been updated in long enough that it appears
+they're not running. When such pipelines are detected, they're re-enqueued from the start.
```

Comment:
Based on reading this, the pipeline itself is not lost in this situation, correct?  There's a means of retrieving it again to place it back in the queue?

Reply:
sort of - there are two persistence models here:
- Redis, used only by Huey
- Postgres, used directly by the broker code and the pipelines we feed to Huey

Redis has the pipeline itself - the data about what huey tasks to run in what order, and what tasks have been run already, etc.
Postgres has the data about the service instance - its current and desired states, and other such data.

When we run into this issue, the data from redis is gone entirely, but the data in postgres is still there.  The `scan_for_stalled_pipelines` job uses the `operation` model in postgres to tell Huey to create a new pipeline in Redis.

Reply:
Okay, that sounds like it answers my question; a pipeline can be reconstructed to be re-run at a later point in time, even though the previous pipeline is gone (which is to be expected)?

Reply:
The answer is yes. :-)